src/SSHDebugPS/Docker/DockerConnection.cs

```diff
@@ -242,5 +244,105 @@ private ICommandRunner GetCommandRunner(DockerContainerTransportSettings setting
             else
                 return new RemoteCommandRunner(settings, OuterConnection);
         }
+
+        private const string dockerVersionCommand = "version";
+        private const string dockerVersionArgs = "--format {{.Server.Os}}";
+
+        private const string dockerInfoCommand = "info";
+        private const string dockerInfoArgs = "--format {{.Driver}}";
+
+        public override List<Process> ListProcesses()
+        {
+            string username;
+            TryGetUsername(out username);
+
+            List<Process> processes;
+            string psErrorMessage;
+            // Try using 'ps' first
+            if (!PSListProcess(username, out psErrorMessage, out processes))
+            {
+                string procErrorMessage;
+                // try using the /proc file system
+                if (!ProcFSListProcess(username, out procErrorMessage, out processes))
+                {
+                    StringBuilder sb = new StringBuilder();
+                    sb.Append(psErrorMessage);
+                    sb.AppendLine(string.Empty);
+                    sb.AppendLine(procErrorMessage);
+
+                    string dockerContainer = DockerHelper.GetDockerOutputString(username, dockerVersionCommand, dockerVersionArgs);
```

Comment:
This won't work if your connecting to a remote SSH connection to get the file list. This method looks like it only runs the command locally.

I think you want to use `ExecuteCommand` but will need to play around with it to see what happens. 

Also you are mixing up `hostname` and `username`. `hostname` is for the `-H` flag to pass to docker if the user  wants to connect to a different docker daemon. `username` is the username associated with the docker session to get the processes for that user.

Reply:
Also think about a better way to do this as the only thing you are adding is the error handling but the rest of the method is copied over. maybe make an interface change that allows you to modify the error message (if necessary) so that there isn't duplicate code here.